qa/sc_evm_test_metamask_related.py

```diff
@@ -0,0 +1,647 @@
+    method = 'transfer(address,uint256)'
+    if static_call:
+        logging.info("Read-only calling {}: testing transfer of ".format(method) +
+              "{} tokens from 0x{} to 0x{}".format(amount, source_account, target_account))
+    else:
+        logging.info("Calling {}: transferring {} tokens from 0x{} to 0x{}".format(method, amount, source_account,
+                                                                            target_account))
+
+    return call_addr_uint_fn(node, smart_contract, contract_address, source_account, target_account, amount,
+                             static_call, generate_block, method)
+
+
+def call_addr_uint_fn(node, smart_contract, contract_address, source_addr, addr, uint, static_call, generate_block,
+                      method):
+    if static_call:
+        res = smart_contract.static_call(node, method, addr, uint,
+                                         fromAddress=source_addr, toAddress=contract_address)
+    else:
+
+        gas = smart_contract.estimate_gas(node, method, addr, uint, fromAddress=source_addr, toAddress=contract_address)
+        res = smart_contract.call_function(node, method, addr, uint,
+                                           fromAddress=source_addr,
+                                           gasLimit=gas, toAddress=contract_address)
+    if generate_block:
+        logging.info("generating next block...")
+        generate_next_blocks(node, "first node", 1)
+    return res
+
+
+def compare_erc20_total_supply(node, smart_contract, contract_address, sender_address, expected_supply):
+    logging.info("Checking total supply of token at 0x{}...".format(contract_address))
+    res = smart_contract.static_call(node, 'totalSupply()', fromAddress=sender_address, toAddress=contract_address)
+    logging.info("Expected supply: '{}', actual supply: '{}'".format(expected_supply, res[0]))
+    assert_equal(res[0], expected_supply)
+    return res[0]
+
+
+def deploy_erc20_smart_contract(node, smart_contract, from_address):
+    logging.info("Deploying smart contract...")
+    logging.info("Estimating gas for deployment...")
+    estimated_gas = smart_contract.estimate_gas(node, 'constructor',
+                                                                fromAddress=from_address)
+    logging.info("Estimated gas is {}".format(estimated_gas))
+    tx_hash, address = smart_contract.deploy(node,
+                                             fromAddress=from_address,
+                                             gasLimit=estimated_gas)
+    logging.info("Generating next block...")
+    generate_next_blocks(node, "first node", 1)
+    tx_receipt = node.rpc_eth_getTransactionReceipt(tx_hash)
+    assert_equal(format_evm(tx_receipt['result']['contractAddress']), format_evm(address))
+    logging.info("Smart contract deployed successfully to address 0x{}".format(address))
+    return address
+
+
+class SCEvmMetamaskTest(SidechainTestFramework):
```

Comment:
Same comments as bloom filter test. In addition, there are duplicated functions from erc20 and erc721 tests